airbyte-scheduler/scheduler-persistence/src/main/java/io/airbyte/scheduler/persistence/job_error_reporter/SentryExceptionHelper.java

```diff
@@ -166,4 +179,130 @@ private static Optional<List<SentryException>> buildJavaSentryExceptions(final S
     return Optional.of(sentryExceptions);
   }
 
+  private static Optional<List<SentryException>> buildNormalizationDbtSentryExceptions(final String stacktrace) {
+    final List<SentryException> sentryExceptions = new ArrayList<>();
+
+    Map<String, String> usefulErrorMap = getUsefulErrorMessageAndTypeFromDbtError(stacktrace);
+
+    // if our errorMessage from the function != stacktrace then we know we've pulled out something
+    // useful
+    if (!usefulErrorMap.get(ERROR_MAP_MESSAGE_KEY).equals(stacktrace)) {
+      final SentryException usefulException = new SentryException();
+      usefulException.setValue(usefulErrorMap.get(ERROR_MAP_MESSAGE_KEY));
+      usefulException.setType(usefulErrorMap.get(ERROR_MAP_TYPE_KEY));
+      sentryExceptions.add(usefulException);
+    }
+
+    if (sentryExceptions.isEmpty())
+      return Optional.empty();
+
+    return Optional.of(sentryExceptions);
+  }
+
+  public static Map<String, String> getUsefulErrorMessageAndTypeFromDbtError(String stacktrace) {
+    // the dbt 'stacktrace' is really just all the log messages at 'error' level, stuck together.
+    // therefore there is not a totally consistent structure to these,
```

Comment:
I think I missed something with regards to the json errors you had mentioned some time ago - were those not useful here for parsing more easily since they would have a [structure](https://docs.getdbt.com/reference/events-logging#structured-logging) to them? Or are we not able to get them?

Reply:
Just saw your other comments that mention the json formatting- are we using them now then? I don't think I understand how they work lol, I'm not seeing any jsons in the tests and it doesn't seem like we're doing any json-y things for parsing.

Reply:
so we are using json formatting and the parsing is happening in `NormalizationAirbyteStreamFactory.java` although really just grabbing out the log level and msg. There's unfortunately not much else of use in there, see these examples:

- `{"code":"Q035","data":{"description":"table model public.midaug_start_users","execution_time":0.1171729564666748,"index":1,"status":"error","total":1},"invocation_id":"69e5b134-b601-4a13-87f8-9525bfc14310","level":"error","log_version":1,"msg":"1 of 1 ERROR creating table model public.midaug_start_users............................................................. [\u001b[31mERROR\u001b[0m in 0.12s]","node_info":{"materialized":"table","node_finished_at":null,"node_name":"midaug_start_users","node_path":"generated/airbyte_incremental/public/midaug_start_users.sql","node_started_at":"2022-08-19T13:42:19.530193","node_status":"compiling","resource_type":"model","type":"node_status","unique_id":"model.airbyte_utils.midaug_start_users"},"pid":16,"thread_name":"Thread-1","ts":"2022-08-19T13:42:19.650138Z","type":"log_line"}`
- `{"code":"Z029","data":{"msg":"  1292 (22007): Truncated incorrect DOUBLE value: 'ABC'"},"invocation_id":"69e5b134-b601-4a13-87f8-9525bfc14310","level":"error","log_version":1,"msg":"  1292 (22007): Truncated incorrect DOUBLE value: 'ABC'","node_info":{},"pid":16,"thread_name":"MainThread","ts":"2022-08-19T13:42:19.748426Z","type":"log_line"}`

Reply:
Ah I see, thanks for explaining